example-plugins/ai_note_titles/protocols/rename_note.py

```diff
@@ -0,0 +1,94 @@
+import json
+
+from canvas_sdk.effects import Effect
+from canvas_sdk.effects.note import Note as NoteEffect
+from canvas_sdk.events import EventType
+from canvas_sdk.protocols import BaseProtocol
+from canvas_sdk.utils.http import Http
+from canvas_sdk.v1.data.command import Command
+from canvas_sdk.v1.data.note import CurrentNoteStateEvent, NoteStates
+from logger import log
+
+
+class Protocol(BaseProtocol):
+    """Protocol that renames Notes when locked using OpenAI and the contents of the Note."""
+
+    RESPONDS_TO: list[str] = [
+        EventType.Name(EventType.NOTE_STATE_CHANGE_EVENT_CREATED),
+    ]
+
+    def compute(self) -> list[Effect]:
+        """This method gets called when an event of the type RESPONDS_TO is fired."""
+        note_id: str | None = self.context.get("note_id")
+
+        if not note_id:
+            log.error("No note ID found in context")
+            return []
+
+        if not self.is_locked_note_event():
+            return []
+
+        new_title = self.get_note_title(note_id)
+
+        if not new_title:
+            return []
+
+        return [NoteEffect(instance_id=note_id, title=new_title).update()]
```

Comment:
I've confirmed everything is working but am unable to get the NoteEffect to actually update a Note title

Could use some help debugging this

Reply:
I just took a look and it looks like the code that updates notes handles titles a little bit strangely. I bet there's a good reason for it, so I'll have to make a bug ticket for someone to dig into. cc: @nmpsilva

Reply:
@nmpsilva how are you getting `self.NOTE_INSTANCE_ID`?

I'm pulling it from `self.context.get("note_id")` which works as expected for `Command.objects.filter(note__id=note_id)` but does not appear to work when creating the Note effect

Reply:
Ah that's just a property I have hardcoded to test, should've been more clear there.
What you're doing should definitely work; I'm gonna debug this today

Reply:
@relm923 Found the issue, and definitely a bug on our side! I think I can push a fix through on one of the next releases.